opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_view_instrument_match.py

```diff
@@ -0,0 +1,110 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from logging import getLogger
+from threading import Lock
+from typing import Iterable, Set
+
+from opentelemetry.sdk._metrics.aggregation import (
+    _convert_aggregation_temporality,
+)
+from opentelemetry.sdk._metrics.measurement import Measurement
+from opentelemetry.sdk._metrics.point import AggregationTemporality, Metric
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.util.instrumentation import InstrumentationInfo
+
+_logger = getLogger(__name__)
+
+
+class _ViewInstrumentMatch:
+    def __init__(
+        self,
+        name: str,
+        unit: str,
+        description: str,
+        aggregation: type,
+        instrumentation_info: InstrumentationInfo,
+        resource: Resource,
+        attribute_keys: Set[str] = None,
+    ):
+        self._name = name
+        self._unit = unit
+        self._description = description
+        self._aggregation = aggregation
+        self._instrumentation_info = instrumentation_info
+        self._resource = resource
+        self._attribute_keys = attribute_keys
+        self._attributes_aggregation = {}
+        self._attributes_previous_point = {}
+        self._lock = Lock()
+
+    def consume_measurement(self, measurement: Measurement) -> None:
+
+        if self._attribute_keys is not None:
+
+            attributes = {}
+
+            for key, value in measurement.attributes.items():
+                if key in self._attribute_keys:
+                    attributes[key] = value
+        else:
+            attributes = measurement.attributes
```

Comment:
this can be None

Reply:
Right, @codeboten fixed this in #2469. :v: